### DISCUSSION TITLE
Are Non-Nunjucks Includes Supported?

### DISCUSSION THREAD

**Discussion by jneander** [11ty/eleventy#2899] [Q&A] [answered]
I searched all over for an existing discussion or issue around this before asking, but could not seem to find an answer. Please forgive me if I missed it.

I've been looking to leverage the JavaScript file support for things that are simply a little too dynamic to manage in Nunjucks, specifically for the inclusion of content fragments, or partials. When I tried to create a `.11ty.js` file in the _includes directory, it was included raw instead of being processed like a Nunjucks file. The same was true for a Markdown file. I can achieve what I need to do through added shortcodes. But it feels like I'm missing something.

Is there a way to configure 11ty to process JavaScript files for includes similarly to how they are processed for layouts?

**Comment by darthmall** [comment, marked as answer]
You can’t use `{% include %}` for anything other than Nunjucks templates and have it be evaluated. The include tag is purely a Nunjucks thing, and Eleventy doesn't have anything to do with it.

My first thought is that what you're looking for is a shortcode: https://www.11ty.dev/docs/shortcodes/

That would let you write some JavaScript functions that returns a string of HTML that you can call from inside your templates.

Another thing that might work is Eleventy’s render plugin: https://www.11ty.dev/docs/plugins/render/

**Reply by jneander** [reply]
This is the answer I was looking for, and then some! I think the `renderFile` shortcode might be just what I need. Thank you!
